src/reference/modules/ROOT/pages/technical_stack.adoc

```diff
@@ -0,0 +1,99 @@
+= Technical stack
+
+== Principles
+
+Technology choices are done component by component, trying to use the right tool for the different needs
+following these criteria:
+
+* Focus on reliability, as Rudder is used on critical production infrastructures, this means:
+
+** For components we develop, choosing languages and tools allowing us to maintain,
+add features and refactor with confidence.
+In this regard, we prefer statically typed languages, with a focus on safety. Our general choices
+are Scala for backend works, Rust for system services and Elm for Web frontend.
+
+** Limit script languages usage to integration/glue code. They are hard to maintain over time and hard to
+deploy in heterogeneous environments.
+
+** Avoid https://en.wikipedia.org/wiki/Not_invented_here[NIH]: use the existing tools that does the job well, especially for things out of our core
+business (web server, data storage, etc.). Use standard system components whenever possible, especially for the agent.
+
+** For components we integrate (programs or libraries), this means choosing a mature and reliable
+option, even more for the agent.
+
+* Agent and relay components need to be able to run on very light systems with low network requirements,
+allowing embedded use cases
+
+* Stay as simple as possible, particularly on the operating side. Users should not have to spend
```

Comment:
I think we need to add here somewhere: "this also means limiting the number of technologies in one given component"